demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/datatable/DataTableTester.java

```diff
@@ -0,0 +1,694 @@
+	 * square brackets with a value of null.
+	 */
+	protected String arrayToString(String[] stringArray) {
+		return Arrays.toString(stringArray).replaceAll("^\\[", "").replaceAll("\\]$", "");
+	}
+
+	/**
+	 * Must build this action using keyDown(key) and keyUp(key) since Action sendKeys behaves differently than
+	 * WebElement sendKeys (unless the browser is phantomjs, then it would need the following script on line 81).
+	 * https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/interactions/Actions.html#sendKeys-java.lang.CharSequence...-
+	 */
+	protected void controlOrCommandClick(Browser browser, String xpath) {
+		Actions commandClickBuilder = browser.createActions();
+		Keys controlOrCommandKey = Keys.CONTROL;
+
+		if (System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("mac")) {
+			controlOrCommandKey = Keys.COMMAND;
+		}
+
+		if ("phantomjs".equals(browser.getName())) {
+			browser.executeScript(
+				"var element = arguments[0]; YUI().use('node-event-simulate', function(Y) { Y.one(element).simulate('click', { metaKey: true }); });",
+				browser.findElementByXpath(xpath));
+		}
+		else {
+			commandClickBuilder.keyDown(controlOrCommandKey).click(browser.findElementByXpath(xpath)).keyUp(
+				controlOrCommandKey);
+
+			Action commandClick = commandClickBuilder.build();
+			commandClick.perform();
+
+			return;
+		}
+
+		return;
+	}
+
+	protected String dataTableContentXpath(String dataContent) {
+		return "(//table[contains(@class,'alloy-data-table')]/tbody/tr/td[contains(.,'" + dataContent + "')])";
+	}
+
+	protected String dataTablePageAndResultsNumberXpath(int pageNumber, String resultsNumber) {
+		return "//div[@class='showcase-example']//ul[@class='pagination']/li/*[contains(.,'Results " + resultsNumber +
+			" of 162 (Page " + pageNumber + " of 17)')]";
+	}
+
+	protected String dataTablePaginationLinkXpath(String whichPaginationActionLink) {
+		return listItemXpath("ul[@class='pagination']", "li") + "/*[contains(@onclick,\"'" + whichPaginationActionLink +
+			"'\")]";
+	}
+
+	protected String getTableRowCustomersXpath(String firstName, String lastName) {
+		return "//tbody/tr[contains(.,'" + firstName + "')][contains(.,'" + lastName +
+			"')]//input[contains(@id,'customers:')]";
+	}
+
+	protected String getTableRowXpath(String firstName, String lastName) {
+		return "//tbody/tr[contains(.,'" + firstName + "')][contains(.,'" + lastName + "')]";
+	}
+
+	protected String labelSelectOptionValueXpath(String labelText, String optionValue) {
+		return "//label[contains(.,'" + labelText + "')]/following-sibling::select/option[@value='" + optionValue +
+			"']";
+	}
+
+	protected void runDataTableLazyTest(String componentName) throws Exception {
+
+		Browser browser = Browser.getInstance();
+		navigateToUseCase(browser, componentName, "lazy");
+
+		testPagesPaginationDefault(browser);
+	}
+
+	protected void runDataTablePaginationTest(String componentName) throws Exception {
```

Comment:
As with https://github.com/stiemannkj1/liferay-faces-alloy/pull/5/commits/9323aba2b9ce98c86cef21d0729992d9cc922cfa#r107022731, please move any methods that are only called by one class into that class.